test/apps/iodemo/io_demo.cc

```diff
@@ -1553,33 +1560,27 @@ class DemoClient : public P2pDemoCommon {
         }
     }
 
-    static void log_disconnect_info(const server_info_t& server_info,
-                                    const char *reason)
-    {
-        bool disconnecting = server_info.conn->is_disconnecting();
+    void disconnect_server(size_t server_index, const char *reason) {
+        server_info_t& server_info = _server_info[server_index];
+        bool disconnecting         = server_info.conn->is_disconnecting();
         std::stringstream ss;
 
-        if (!disconnecting) {
+        if (disconnecting) {
```

Comment:
why changed?

Reply:
i think it should be inversed condition

Reply:
it was a bug.
if we are already disconnecting from this server, we should say ""not disconnecting .. because ..."

Reply:
ok, right